### PR TITLE
Adds Retry Logic for Requests

### PR DESCRIPTION
**Pull request by bretterer**
This handles the HTTP Status code 429 for Too Many Request and the retry logic around it.

**New Config Items**
```
okta:
  client:
    requestTimeout: 0 # time to wait for resolved client request (includes retries), <= 0 means "no timeout"
    rateLimit:
      maxRetries: 2 # Use 0 to disable max retry breaker
```

### CODE REVIEW

**Comment by quantumew** [comment]
Hopefully you don't mind me commenting. This won't work if there is a request body, right? Unless I am missing something this doesn't reset the request body, which is read on each request. Also, it is possible in its current state to be closed before the next retry, I believe. I think the request body needs to be wrapped in a `NopCloser` to prevent closing and reset on every retry. Thanks for adding this. Feel free to close #92.

**Comment by bretterer** [comment]
@quantumew You are correct, it is still a WIP.  Just wanted to get something up for the moment.

I am working now on taking your PR and merging it in with the framework I have here.  There were just some items that I needed to add in first to make sure the config options we had were available

**Comment by bretterer** [comment]
@quantumew how does this look to you?  

I had to update your code a bit because it seems like it got rid of the SSWS headers and also changed the API for `NewRequest` which is used everywhere in the code. 

There are still a few requirements we have that I will be tweaking, but does this method solve your needs?

File: okta/requestExecutor.go [hunk 1]
```diff
@@ -82,40 +85,107 @@ func (re *RequestExecutor) NewRequest(method string, url string, body interface{
...
+	_, err := io.Copy(ioutil.Discard, io.LimitReader(body, 4096))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func backoffPause(retryCount int32, response *http.Response) error {
+	if response.StatusCode == http.StatusTooManyRequests {
+		resetLimit, _ := strconv.Atoi(response.Header.Get("X-Rate-Limit-Reset"))
```
[the remark this answers is not in the document]
Comment (bretterer) [reply okta/requestExecutor.go]: Handled

File: okta/requestExecutor.go [hunk 2]
```diff
@@ -82,40 +85,107 @@ func (re *RequestExecutor) NewRequest(method string, url string, body interface{
...
+		requestDate, _ := time.Parse("Mon, 02 Jan 2006 15:04:05 Z", response.Header.Get("Date"))
```
[the remark this answers is not in the document]
Comment (bretterer) [reply okta/requestExecutor.go]: Handled

**Pull request merged**